File: src/main/java/com/divide2/auth/service/impl/UserServiceImpl.java
package com.divide2.auth.service.impl;

import com.divide2.auth.model.User;
import com.divide2.auth.repository.UserRepository;
import com.divide2.auth.service.IUserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;

@Service
public class UserServiceImpl implements IUserService {

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private PasswordEncoder bCryptPasswordEncoder;

    @Override
    public User getByUsername(String username) {
        return userRepository.getByUsername(username);
    }

    @Override
    public void save(User user) {
        user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
        user.setCreateDateTime(LocalDateTime.now());
        userRepository.save(user);
    }
}

File: src/main/kotlin/com/divide2/auth/user/UserServiceImpl.kt
package com.bvvy.grocery.auth.user

import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service

@Service
class UserServiceImpl(val userRepository: UserRepository,val passwordEncoder: PasswordEncoder) : UserService {

    override fun save(user: User) {
        val existsUser: User? = loadByPhone(user.phone)
        if (existsUser != null) {
//            throw AlreadyExistsException()
        }
        user.password = passwordEncoder.encode(user.password)
        userRepository.save(user)
    }

    override fun oauthSave(user: User) {
        userRepository.save(user)
    }

    override fun loadByPhone(phone: String): User? {
        return userRepository.findByPhone(phone)
    }


}